.github/workflows/nightly-emscripten.yml

```diff
@@ -0,0 +1,96 @@
+name: Emscripten nightly build + solc-bin push
+
+on:
+  schedule:
+    # Run once a day, at midnight
+    - cron: '0 0 * * *'
+
+env:
+  TARGET_BRANCH: master
+  COMMITTER_NAME: pull-soljson action
+  COMMITTER_EMAIL: builds@ethereum.org
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  build-emscripten-nightly:
+    runs-on: ubuntu-latest
+    outputs:
+      nightly-version: ${{ env.NIGHTLY_VERSION }}
+      matching-nightlies-in-the-repo: ${{ env.MATCHING_NIGHTLIES_IN_THE_REPO }}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: 'ethereum/solidity'
+          ref: 'develop'
+          path: 'solidity/'
+
+      - name: Clone solc-bin repository without checking out a working copy
+        run: |
+          git clone --no-checkout --branch "$TARGET_BRANCH" "https://github.com/${GITHUB_REPOSITORY}.git" solc-bin/
+
+          # For some reason git stages all files for deletion when you use --no-checkout
+          cd solc-bin/
+          git reset HEAD --quiet
+
+      - name: Check if there's already a nightly with the same date or commit ID
+        run: |
+          cd solidity/
+          solidity_version=$("scripts/get_version.sh")
+          last_commit_timestamp=$(git log -1 --date=iso --format=%ad HEAD)
+          last_commit_date=$(date --date="$last_commit_timestamp" --utc +%Y.%-m.%-d)
+          last_commit_hash=$(git rev-parse --short=8 HEAD)
+
+          cd ../solc-bin/
+          matching_nightlies_in_the_repo="$(
+            git ls-files "bin/soljson-v${solidity_version}-nightly.${last_commit_date}+commit.*.js";
+            git ls-files "bin/soljson-v${solidity_version}-nightly.*+commit.${last_commit_hash}.js"
+          )"
+          nightly_version="v${solidity_version}-nightly.${last_commit_date}+commit.${last_commit_hash}"
```

Comment:
I went back to constructing nightly version manually (rather than using `get_nightly_version.sh`) because now I need the commit hash and date anyway and I have all the information needed to do it at hand. The script no longer adds much value, and just introduces the possibility of things going out of sync.

Should I leave it like that (and perhaps remove `get_nightly_version.sh`) or should I add another script (say, `find_matching_nightlies $date $commit`) in `solidity` repo alongside it?

Reply:
Somehow I feel that the whole version juggling here (especially if the version is already there) could be better placed inside the update script, or at least in a nodejs script that uses the same utilities.

Reply:
After giving it some thought, how about moving the regex matching from the `update` script into a separate one that would split binary name into components, e.g.:
```
$ ./split-release-binary-name.sh "soljson-v0.6.1-nightly.2019.12.20+commit.ece6463f.js"
```
```
{
    "name": "soljson",
    "version": "0.6.1",
    "type": "nightly",
    "timestamp": "2019.12.20",
    "commit": "ece6463f",
    "extension": "js"
}
```
And we could also replace `get_nightly_version.sh` with one that does the reverse - gets this dict on stdin and prints the name. If there's nothing on stdin it would get current values from `git` and `date`.

JSON output it should make it easy to use `split-release-binary-name.sh` both in nodejs scripts and in the shell (with the help of `jq`).

Then, in this action, we could run `git ls-files` to get a list of all files in `bin/`, pass each name through the script and run the result through `jq` to filter out commits and dates that do not match. Could even be a one-liner, though we'd probably want to split it for readability.

Reply:
Hm, maybe it's perfectly fine as it is :)

Reply:
Maybe the description above makes it look more complex that it would actually be. But yeah, nothing beats the simplicity of just not adding more stuff so that's fine too :)